### PR TITLE
[wasm] Tweak set timeout

### PR DESCRIPTION
**Pull request by maraf**
- Cache method lookup for TimerQueue.TimeoutCallback and ThreadPool.Callback.
- Remove unused parameter "id" from timer related functions.

### CODE REVIEW

**Comment by maraf** [comment]
Some tests are failing. I'm not sure it's caused by my changes.

File: src/mono/wasm/runtime/scheduling.ts [hunk 1]
```diff
@@ -52,9 +52,9 @@ export function schedule_background_exec(): void {
 }
 
 let lastScheduledTimeoutId: any = undefined;
-export function mono_set_timeout(timeout: number, id: number): void {
+export function mono_set_timeout(timeout: number): void {
     function mono_wasm_set_timeout_exec() {
```
Comment (kg) [inline src/mono/wasm/runtime/scheduling.ts:56]: you don't really need this wrapper anymore, I think?

Comment (pavelsavara) [reply src/mono/wasm/runtime/scheduling.ts:56]: Just make sure you rename it in the unit test where we count it by name. Also in rollup config.

Comment (maraf) [reply src/mono/wasm/runtime/scheduling.ts:56]: Even with the tests in mind, I can directly pass the `cwraps.mono_set_timeout_exec` to `setTimeout`. The only other place where it is registered to `setTimeout` is from `prevent_timer_throttling` and that won't effect the test.

Comment (maraf) [reply src/mono/wasm/runtime/scheduling.ts:56]: The name is the only way to compare a callback (I'm able to access `cwraps.mono_set_timeout_exec` from test and not comparing the callback could make the test fragile).

**Review by lambdageek — APPROVED** [review]

**Review by pavelsavara — APPROVED** [review]

File: src/mono/wasm/runtime/scheduling.ts (hunk 1 above)
Comment (kg) [reply src/mono/wasm/runtime/scheduling.ts:56]: If you want to compare delegates for identity you could use them as ```Map``` keys or put a property on them. Though I think using the name for test purposes is fine

**Pull request merged**